src/rewrites.jl

```diff
@@ -0,0 +1,139 @@
+        else
+            $setup
+            $local_copy[!, $target_column] .= missing
+            $sdf[!, $target_column] .= $RHS
+            $local_copy |> $teardown
+        end
+    end |> esc
+end
+
+function rewrite(::Val{:replace}, command::Command)
+    gc = generate_command(command; options=[:single_argument, :variables, :ifable, :replace_variables, :vectorize, :assignment])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    target_column = get_LHS(command.arguments[1])
+    LHS, RHS = split_assignment(arguments[1])
+    third_vector = gensym()
+    bitmask = build_bitmask(local_copy, vectorize_function_calls(replace_variable_references(local_copy, command.condition)))
+    quote
+        if !($target_column in names($df))
+            ArgumentError("Column \"$($target_column)\" does not exist in $(names($df))") |> throw
+        else
+            $setup
+            if eltype($RHS) != eltype($sdf[!, $target_column])
+                local $third_vector = Vector{eltype($RHS)}(undef, nrow($local_copy))
+                $third_vector[$bitmask] .= $RHS
+                $third_vector[.!$bitmask] .= $local_copy[!, $target_column][.!$bitmask]
+                $local_copy[!, $target_column] = $third_vector
+            else
+                $sdf[!, $target_column] .= $RHS
+            end
+            $local_copy |> $teardown
+        end
+    end |> esc
+end
+
+function rewrite(::Val{:collapse}, command::Command)
+    gc = generate_command(command; options=[:variables, :ifable, :replace_variables, :vectorize, :assignment])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    by_cols = get_by(command)
+    if isnothing(by_cols)
+        combine_epxression = Expr(:call, :combine, sdf, build_assignment_formula.(command.arguments)...)
+    else
+        combine_epxression = Expr(:call, :combine, gdf, build_assignment_formula.(command.arguments)...)
+    end
+    quote
+        $setup
+        if isnothing($by_cols)
+            $combine_epxression |> $teardown
+        else
+            local $gdf = groupby($sdf, $by_cols)
+            $combine_epxression |> $teardown
+        end
+    end |> esc
+end
+
+function rewrite(::Val{:keep}, command::Command)
+    gc = generate_command(command; options=[:variables, :ifable])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    quote
+        $setup
+        $sdf[!, isempty($(command.arguments)) ? eval(:(:)) : collect($command.arguments)]  |> $teardown
+    end |> esc
+end
+
+function rewrite(::Val{:drop}, command::Command)
+    gc = generate_command(command; options=[:variables, :ifable])
+    (; df, local_copy, sdf, gdf, setup, teardown, arguments) = gc
+    if isnothing(command.condition)
+        return quote
+            $setup
+            select($local_copy, Not(collect($(command.arguments)))) |> $teardown
+        end |> esc
+    end 
+    bitmask = build_bitmask(local_copy, :(!($command.condition)))
```

Comment:
Here again, I might get something wrong but wouldn't it be enough to refer $sdf.

Reply:
For drop, we need the rows for which the condition is NOT true.